app/models/user.rb

```diff
@@ -101,6 +101,10 @@ def pending_profile
     profiles.gpo_verification_pending.order(created_at: :desc).first
   end
 
+  def personal_key_generated_at
+    encrypted_recovery_code_digest_generated_at || active_profile&.verified_at
```

Comment:
As @mitchellhenke [pointed out in Slack](https://gsa-tts.slack.com/archives/C0NGESUN5/p1676311278523799?thread_ts=1676308836.400049&cid=C0NGESUN5), the password reactivation profile will not be considered "active", so we need to fall back to [`password_reset_profile`](https://github.com/18F/identity-idp/blob/4deb7e148a151af4cf4c869420b0b6f60272d7dd/app/decorators/user_decorator.rb#L73-L78) for the account reactivation flow.